src/core/tokenizer.c

```diff
@@ -149,31 +154,83 @@ tokenset_resolve(tokenset_t *tset, nxs_index_t *idx, bool stage)
 	}
 }
 
+
+void		char_from_uchar_slice(char *, UChar* , int32_t , int32_t);
+void
+char_from_uchar_slice(char* buf, UChar* str, int32_t start, int32_t end)
+{
+	UChar   savedEndChar;
+
+	savedEndChar = str[end];
+	str[end] = 0;
+	u_austrcpy(buf, str+start);
+	str[end] = savedEndChar;
+}
+
+
 /*
- * tokenize: split the text into tokens separated by a whitespace
- * and pass the token through the filter pipeline.
+ * tokenize: Uses ICU segmentation UBRK_WORD
+ * See: https://unicode.org/reports/tr29/
  */
 tokenset_t *
-tokenize(filter_pipeline_t *fp, const char *text, size_t text_len __unused)
+tokenize(filter_pipeline_t *fp, const char *text, size_t text_len, const char *lang)
```

Comment:
I would argue that you should pass the params to the tokenizer. It is likely that there will be more params later, so:
`tokenize(filter_pipeline_t *fp, nxs_params_t *params, const char *text, size_t text_len)`

Reply:
yes I prefer to pass the param actually! in the future we may have more params